File: Hardened_Runtime_Check.py
#!/usr/bin/env python3
import sys
import os
import subprocess

d = os.listdir("/Applications/")

for each in d:
	if each.endswith('.app'):
		apath = "/Applications/%s/Contents/MacOS" % each
		if os.access(apath, os.R_OK):
			p = os.listdir(apath)
			for item in p:
				apath2 = apath + "/" + item
				try:
					x = subprocess.getstatusoutput('codesign --display --verbose "%s"' % apath2)
					if (('flags' in x[1]) and ('runtime' in x[1])):
						print("\033[92m[+] Hardened runtime enabled in: %s\033[0m" % apath2)
					elif (('flags' in x[1]) and ('runtime' not in x[1])):
						print("\033[33m[-] Hardened runtime NOT enabled in: %s\033[0m" % apath2)
					else:
						pass
				except:
					pass



c = os.listdir("/usr/local/bin")

for binary in c:
	pth = "/usr/local/bin/%s" % binary
	try:
		if os.access(pth, os.R_OK):
			z = subprocess.getstatusoutput('codesign --display --verbose "%s"' % pth)
			if (('flags' in z[1]) and ('runtime' in z[1])):
				print("\033[92m[+] Hardened runtime enabled in: %s\033[0m" % pth)
			elif (('flags' in z[1]) and ('runtime' not in z[1])):
				print("\033[33m[-] Hardened runtime NOT enabled in: %s\033[0m" % pth)
			else:
				pass
	except:
		pass




b = os.listdir("/usr/sbin")

for bin in b:
	pth2 = "/usr/sbin/%s" % bin
	try:
		if os.access(pth2, os.R_OK):
			y = subprocess.getstatusoutput('codesign --display --verbose "%s"' % pth2)
			if (('flags' in y[1]) and ('runtime' in y[1])):
				print("\033[92m[+] Hardened runtime enabled in: %s\033[0m" % pth2)
			elif (('flags' in y[1]) and ('runtime' not in y[1])):
				print("\033[33m[-] Hardened runtime NOT enabled in: %s\033[0m" % pth2)
			else:
				pass
	except:
		pass




a = os.listdir("/usr/bin")

for file in a:
	pth3 = "/usr/bin/%s" % file
	try:
		q = subprocess.getstatusoutput('codesign --display --verbose "%s"' % pth3)
		if (('flags' in q[1]) and ('runtime' in q[1])):
			print("\033[92m[+] Hardened runtime enabled in: %s\033[0m" % pth3)
		elif (('flags' in q[1]) and ('runtime' not in q[1])):
			print("\033[33m[-] Hardened runtime NOT enabled in: %s\033[0m" % pth3)
		else:
			pass
	except:
		pass 

File: HardenedRuntimeCheck.swift
import Foundation
import Cocoa

let green = "\u{001B}[0;32m"
let yellow = "\u{001B}[0;33m"
let colorend = "\u{001B}[0;0m"

func AppDirCheck(){

    do {
        let fileMan = FileManager.default
        let appdir = try fileMan.contentsOfDirectory(atPath: "/Applications")
        print("===========================/Applications Directory Check=================================")
        for dir in appdir{
            if dir.hasSuffix(".app"){
                let app_path = "/Applications/\(dir)/Contents/MacOS"

                do {
                    let contents = try fileMan.contentsOfDirectory(atPath: app_path)

                    for item in contents {
                        let path2 = "\(app_path)/\(item)"

                        let proc = Process()
                        proc.launchPath = "/usr/bin/codesign"

                        let args : [String] = ["-d", "--verbose", "\(path2)"]

                        proc.arguments = args
                        let pipe = Pipe()

                        proc.standardError = pipe
                        proc.launch()

                        let results = pipe.fileHandleForReading.readDataToEndOfFile()
                        let out = String(data: results, encoding: String.Encoding.utf8)!


                        if out.contains("flags") && out.contains("runtime"){
                                print("\(green)[-] Hardened runtime enabled in: \(path2)\(colorend)")
                        }
                        else if out.contains("flags") && !(out.contains("runtime")){
                                print("\(yellow)[-] Hardened runtime NOT enabled in: \(path2)\(colorend)")
                        }
                        else {

                        }


                    }
                } catch {

                }
            }
        }
    }
    catch {

    }
    print("")

}

func UtiltiesDirCheck(){

    do {
    let fileMan = FileManager.default
    let appdir = try fileMan.contentsOfDirectory(atPath: "/System/Applications/Utilities/")
    print("===========================/System/Applications/Utilities/ Directory Check=================================")
    for dir in appdir{
        if dir.hasSuffix(".app"){
            let app_path = "/System/Applications/Utilities/\(dir)/Contents/MacOS"

            do {
                let contents = try fileMan.contentsOfDirectory(atPath: app_path)

                for item in contents {
                    let path2 = "\(app_path)/\(item)"

                    let proc = Process()
                    proc.launchPath = "/usr/bin/codesign"
                    let args : [String] = ["-d", "--verbose", "\(path2)"]
                    proc.arguments = args
                    let pipe = Pipe()

                    proc.standardError = pipe
                    proc.launch()

                    let results = pipe.fileHandleForReading.readDataToEndOfFile()
                    let out = String(data: results, encoding: String.Encoding.utf8)!

                    if out.contains("flags") && out.contains("runtime"){
                            print("\(green)[-] Hardened runtime enabled in: \(path2)\(colorend)")
                    }
                    else if out.contains("flags") && !(out.contains("runtime")){
                            print("\(yellow)[-] Hardened runtime NOT enabled in: \(path2)\(colorend)")
                    }
                    else {

                    }


                }
            } catch {

            }
        }
    }
}
catch {

}
    print("")

}

func LocalBinCheck(){

    do {
        let fileMan = FileManager.default
        let appdir = try fileMan.contentsOfDirectory(atPath: "/usr/local/bin/")

        print("===========================/usr/local/bin Directory Check=================================")

        for bin in appdir{

                do {
                    let bin_path = "/usr/local/bin/\(bin)"

                        let proc = Process()
                        proc.launchPath = "/usr/bin/codesign"
                        let args : [String] = ["-d", "--verbose", "\(bin_path)"]
                        proc.arguments = args
                        let pipe = Pipe()

                        proc.standardError = pipe
                        proc.launch()

                        let results = pipe.fileHandleForReading.readDataToEndOfFile()
                        let out = String(data: results, encoding: String.Encoding.utf8)!

                        if out.contains("flags") && out.contains("runtime"){
                                print("\(green)[-] Hardened runtime enabled in: \(bin_path)\(colorend)")
                        }
                        else if out.contains("flags") && !(out.contains("runtime")){
                                print("\(yellow)[-] Hardened runtime NOT enabled in: \(bin_path)\(colorend)")
                        }
                        else {

                        }

                } catch {

                }
        }
    }
    catch {

    }
    print("")

}

func sbinDirCheck(){
    do {
        let fileMan = FileManager.default
        let appdir = try fileMan.contentsOfDirectory(atPath: "/usr/sbin/")

        print("===========================/usr/sbin Directory Check=================================")

        for bin in appdir{

                do {
                    let bin_path = "/usr/sbin/\(bin)"

                        let proc = Process()
                        proc.launchPath = "/usr/bin/codesign"
                        let args : [String] = ["-d", "--verbose", "\(bin_path)"]
                        proc.arguments = args
                        let pipe = Pipe()

                        proc.standardError = pipe
                        proc.launch()

                        let results = pipe.fileHandleForReading.readDataToEndOfFile()
                        let out = String(data: results, encoding: String.Encoding.utf8)!

                        if out.contains("flags") && out.contains("runtime"){
                                print("\(green)[-] Hardened runtime enabled in: \(bin_path)\(colorend)")
                        }
                        else if out.contains("flags") && !(out.contains("runtime")){
                                print("\(yellow)[-] Hardened runtime NOT enabled in: \(bin_path)\(colorend)")
                        }
                        else {

                        }

                } catch {

                }
        }
    }
    catch {

    }
    print("")
}

func binDirCheck(){

    do {
        let fileMan = FileManager.default
        let appdir = try fileMan.contentsOfDirectory(atPath: "/usr/bin/")

        print("===========================/usr/bin Directory Check=================================")

        for bin in appdir{

                do {
                    let bin_path = "/usr/bin/\(bin)"

                        let proc = Process()
                        proc.launchPath = "/usr/bin/codesign"
                        let args : [String] = ["-d", "--verbose", "\(bin_path)"]
                        proc.arguments = args
                        let pipe = Pipe()

                        proc.standardError = pipe
                        proc.launch()

                        let results = pipe.fileHandleForReading.readDataToEndOfFile()
                        let out = String(data: results, encoding: String.Encoding.utf8)!

                        if out.contains("flags") && out.contains("runtime"){
                                print("\(green)[-] Hardened runtime enabled in: \(bin_path)\(colorend)")
                        }
                        else if out.contains("flags") && !(out.contains("runtime")){
                                print("\(yellow)[-] Hardened runtime NOT enabled in: \(bin_path)\(colorend)")
                        }
                        else {

                        }

                } catch {

                }
        }
    }
    catch {

    }
    print("")

}

print("===> Checking common app/bin locations for hardened runtime...")
AppDirCheck()
UtiltiesDirCheck()
LocalBinCheck()
sbinDirCheck()
binDirCheck()

print("[+] DONE!")
